core/src/main/java/org/apache/accumulo/core/data/AbstractId.java

```diff
@@ -30,7 +30,9 @@
   private final String canonical;
 
   protected AbstractId(final String canonical) {
-    this.canonical = Objects.requireNonNull(canonical, "canonical cannot be null");
+    if (canonical == null || canonical.trim().isEmpty())
+      throw new IllegalArgumentException("Id string provided can't be empty or null.");
```

Comment:
Since the parameter name is "canonical", it'd be good to use that word when referencing the parameter.

```suggestion
      throw new IllegalArgumentException("canonical id provided can't be empty or null.");
```